google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/HeaderInterceptor.java

```diff
@@ -137,6 +146,26 @@ private TagContext getTagContext(Metadata headers, String method) {
         LOGGER.log(LEVEL, "Error parsing google cloud resource header: " + googleResourcePrefix);
       }
     }
-    return getTagContext(method, projectId, instanceId, databaseId);
+    return new SpannerProperties(projectId, instanceId, databaseId);
+  }
+
+  private TagContext getTagContext(String method, SpannerProperties spannerProperties) {
+    return TAGGER
+        .currentBuilder()
+        .putLocal(PROJECT_ID, TagValue.create(spannerProperties.projectId))
+        .putLocal(INSTANCE_ID, TagValue.create(spannerProperties.instanceId))
+        .putLocal(DATABASE_ID, TagValue.create(spannerProperties.databaseId))
+        .putLocal(METHOD, TagValue.create(method))
+        .build();
+  }
+
+  private Attributes getMetricAttributes(String method, SpannerProperties spannerProperties) {
+    AttributesBuilder attributesBuilder = Attributes.builder();
+    attributesBuilder.put("database", spannerProperties.databaseId);
+    attributesBuilder.put("instance_id", spannerProperties.instanceId);
```

Comment:
Agree. But here we are migrating existing metrics, these attributes were already present. So if we remove these attributes now, it will be a breaking change for customers. We will keep this in mind with new metrics